test/functional/role_test.rb

```diff
@@ -0,0 +1,72 @@
+require File.join(File.dirname(__FILE__), 'test_helper')
+
+describe 'role' do
+  describe 'clone' do
+    before do
+      @cmd = %w(role clone)
+    end
+
+    it 'should print error on missing --id' do
+      params = ['--new-name=zzz']
+
+      expected_result = CommandExpectation.new
+      expected_result.expected_err =
+        ['Could not clone the user role:',
+         "  Missing arguments for 'id'",
+         ''].join("\n")
+      expected_result.expected_exit_code = HammerCLI::EX_USAGE
+
+      api_expects_no_call
+
+      result = run_cmd(@cmd + params)
+      assert_cmd(expected_result, result)
+    end
+
+    it 'should print error on missing --new-name' do
+      params = ['--id=1']
+
+      expected_result = usage_error_result(
+        @cmd,
+        'Option --new-name is required',
+        'Could not clone the user role')
+
+      api_expects_no_call
+
+      result = run_cmd(@cmd + params)
+
+      assert_cmd(expected_result, result)
+    end
+
+    it 'should clone a role by id' do
+      params = ['--id=1', '--new-name=zzz']
+
+      api_expects(:roles, :clone, 'Clone role') do |par|
+        par['id'] == '1' && par['new_name'] == 'zzz'
```

Comment:
I think we are good here, the api export is already updated, I just did not push the new version of api to [theforeman/foreman#4267](https://github.com/theforeman/foreman/pull/4267) for some reason - fixed now. The test works as expected, role should be cloned when `--id` and `--new-name` are supplied, where `new-name` is a [name of the role that is being created](https://github.com/theforeman/hammer-cli-foreman/pull/283#issuecomment-280292557).